### PR TITLE
Validate callback URL length

### PR DESCRIPTION
**Pull request by rodrigozhou**
## What changed?

Validate callback URL length.

## Why?

Prevent very long URL.

## How did you test it?

Added test.

## Is hotfix candidate?

No.

### CODE REVIEW

File: service/frontend/workflow_handler.go [hunk 1]
```diff
@@ -415,22 +418,11 @@ func (wh *WorkflowHandler) StartWorkflowExecution(ctx context.Context, request *
...
-			if err != nil {
-				return nil, status.Errorf(codes.InvalidArgument, "invalid url: %v", err)
-			}
-			if !(u.Scheme == "http" || u.Scheme == "https") {
-				return nil, status.Errorf(codes.InvalidArgument, "invalid url scheme for url: %v", u)
-			}
-			// TODO: check in dynamic config that address is valid and that http is only accepted if "insecure" is
-			// allowed for address.
-			// TODO: validate callback URL length against dynamic config.
-		}
```
Comment (rodrigozhou) [inline service/frontend/workflow_handler.go:433]: Created a function to resolve cyclomatic complexity lint error.

File: service/frontend/workflow_handler.go [hunk 2]
```diff
@@ -4121,6 +4113,46 @@ func (wh *WorkflowHandler) validateWorkflowIdReusePolicy(
...
+				return status.Errorf(codes.InvalidArgument, "invalid url: %v", err)
+			}
+			if !(u.Scheme == "http" || u.Scheme == "https") {
+				return status.Errorf(codes.InvalidArgument, "invalid url scheme for url: %v", u)
+			}
+			// TODO: check in dynamic config that address is valid and that http is only accepted
+			// if "insecure" is allowed for address.
+
+		default:
+			return status.Error(codes.Unimplemented, fmt.Sprintf("unimplemented callback type: %T", cb))
```
Comment (rodrigozhou) [inline service/frontend/workflow_handler.go:4150]: Before, the validation was just ignoring callbacks that are not of Nexus type. I changed to return an error if it there is a callback of a unknown type.
@bergundy Let me know what you think.

File: service/frontend/workflow_handler.go [hunk 3]
```diff
@@ -4121,6 +4113,46 @@ func (wh *WorkflowHandler) validateWorkflowIdReusePolicy(
...
+		return status.Error(
+			codes.InvalidArgument,
+			"attaching workflow callbacks is disabled for this namespace",
+		)
+	}
+
+	for _, callback := range callbacks {
+		switch cb := callback.GetVariant().(type) {
+		case *commonpb.Callback_Nexus_:
+			if len(cb.Nexus.GetUrl()) > wh.config.CallbackURLMaxLength() {
```
Comment (tdeebswihart) [inline service/frontend/workflow_handler.go:4130]: nit: If cb.Nexus isn't nil you could use `cb.Nexus.Url` directly

File: service/frontend/workflow_handler.go [hunk 4]
```diff
@@ -4121,6 +4113,46 @@ func (wh *WorkflowHandler) validateWorkflowIdReusePolicy(
...
+						wh.config.CallbackURLMaxLength(),
+					),
+				)
+			}
+			u, err := url.Parse(cb.Nexus.GetUrl())
+			if err != nil {
+				return status.Errorf(codes.InvalidArgument, "invalid url: %v", err)
+			}
+			if !(u.Scheme == "http" || u.Scheme == "https") {
+				return status.Errorf(codes.InvalidArgument, "invalid url scheme for url: %v", u)
```
Comment (tdeebswihart) [inline service/frontend/workflow_handler.go:4144]:
```suggestion
				return status.Errorf(codes.InvalidArgument, "invalid url: unknown scheme %v", u)
```
				
This keeps it consistent with the other errors

File: service/frontend/workflow_handler.go (hunk 3 above)
Comment (rodrigozhou) [reply service/frontend/workflow_handler.go:4130]: cc: @bergundy

**Review by bergundy — APPROVED** [review]

File: service/frontend/workflow_handler.go (hunk 3 above)
Comment (bergundy) [reply service/frontend/workflow_handler.go]: I think the `type` ensures that `cb.Nexus` isn't nil but I'm not 100% sure.

File: service/frontend/workflow_handler.go (hunk 2 above)
Comment (bergundy) [reply service/frontend/workflow_handler.go]: Yes, thanks!

**Pull request merged**